Assignment 2.Rmd

```diff
@@ -85,16 +86,62 @@ plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender, edge.widt
 
 ## Part II
 
-In Part II your task is to [look up](http://igraph.org/r/) in the igraph documentation and create a graph that sizes the student vertices in terms of the number of disciplinary actions they have recieved, and the teachers in terms of the number of disciplinary actions they have given out.
+#In Part II your task is to [look up](http://igraph.org/r/) in the igraph documentation and create a graph that sizes the student vertices in terms of the number of disciplinary actions they have recieved, and the teachers in terms of the number of disciplinary actions they have given out.
 
 
+```{r}
+library("dplyr")
+library("igraph")
+C1 <- count(D1, stid)
+C2 <- count(D1, tid)
+names(C1) <- c("id", "count")
+names(C2) <- c("id", "count")
+C3 <- rbind(C1, C2)
+VERTEX2 <- full_join(VERTEX, C3)
+plot(g, layout = layout.fruchterman.reingold, vertex.color = VERTEX$gender, edge.width = EDGE$count, vertex.size = VERTEX2$count)
 
+```
 ## Part III
 
 Now practice with data from our class. Please create a **person-network** with the data set hudk4050-classes.csv. To create this network you will need to create a person-class matrix using the tidyr functions and then create a person-person matrix using `t()`. You will then need to plot a matrix rather than a data frame using igraph.
 
 Once you have done this, also [look up](http://igraph.org/r/) how to generate the following network metrics: betweeness centrality and dregree. **Who is the most central person in the network?**
 
+```{r}
+library("tidyverse")
+classes <- read.csv("D:/Github/4050/assignment2/assignment2/class1.csv")
+class1 <- select(classes, V18:V26) %>% slice(-c(1:3))
+class2 <- unite(class1, V18, V19, col = "V2", sep = " ")
+class2 <- select(class2, c(V2, V21:V26))
+class2 <- slice(class2, -1)
+t1 <- t(class2)
+
+
+colnames(class2) = c("name", "class1", "class2", "class3", "class4", "class5", "class6")
+
+aaa <- gather(class2, st_name, class, 2:7)
+aaa <- select(aaa, -2)
+aaa <- mutate(aaa, count=1)
+aaa1 <- as.data.frame(aaa)
+aaa2 <- unique(aaa1)
+aaa3 <- spread(aaa2, key = class,value = count)
+
+aaa4 <- aaa3[,-1]
+rownames(aaa4) <- aaa3[,1]
+aaa5 <- as.matrix(aaa4)
+aaa5[is.na(aaa5)] <-0
+#student to class matrix
+class(aaa5) <- "numeric"
+aaa6 <- aaa5 %*% t(aaa5)
+#student to student martrix
+diag(aaa6) <- NA
+library("igraph")
+gg <- graph.adjacency(aaa6, mode = "undirected")
+plot(gg, layout = layout.fruchterman.reingold)
+#graph of student network
```

Comment:
Very nice work, clear and concise data cleaning and plotting. Consider adding more attributes so that the plot shows the number of common classes shared by people. 
One thing left is to find out the most central person in the network.